api/client.go

```diff
@@ -281,12 +281,13 @@ func (c *Client) AbortCurrentUpgrade() error {
 }
 
 // FindTools returns a List containing all tools matching the specified parameters.
-func (c *Client) FindTools(majorVersion, minorVersion int, series, arch string) (result params.FindToolsResult, err error) {
+func (c *Client) FindTools(majorVersion, minorVersion int, series, arch, agentStream string) (result params.FindToolsResult, err error) {
 	args := params.FindToolsParams{
 		MajorVersion: majorVersion,
 		MinorVersion: minorVersion,
 		Arch:         arch,
 		Series:       series,
+		AgentStream:  agentStream,
```

Comment:
Can you check the facade version here and error if it's 1 and agentStream is set? Otherwise it'll just be silently ignored.